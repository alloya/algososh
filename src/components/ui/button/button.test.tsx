import { Button } from "./button";
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from "@testing-library/react";
const buttonText = 'test';
const clickFn = () => {
  alert(buttonText);
}

describe('Button component renders correctly', () => {
  it('with text', () => {
    const button = renderer.create(<Button text={buttonText} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('without text', () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('with disabled attribute', () => {
    const button = renderer.create(<Button disabled={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('with loaded state', () => {
    const button = renderer.create(<Button isLoader={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });
  
  it('with onClick action', () => {
    window.alert = jest.fn();
    render(<Button onClick={clickFn} />);
    const button = screen.getByRole("button");
    fireEvent.click(button);
    expect(window.alert).toHaveBeenCalledWith(buttonText);
  });
})