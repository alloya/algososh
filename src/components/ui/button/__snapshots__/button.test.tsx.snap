// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly with disabled attribute 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component renders correctly with loaded state 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button component renders correctly with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test
  </p>
</button>
`;

exports[`Button component renders correctly without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
